pkg/utils/utils.go

```diff
@@ -53,12 +54,22 @@ func RetrieveVcConfigSecret(params map[string]interface{}, logger logrus.FieldLo
 
 	ns := "kube-system"
 	secretApis := clientset.CoreV1().Secrets(ns)
-	vsphere_secret := "vsphere-config-secret"
-	secret, err := secretApis.Get(vsphere_secret, metav1.GetOptions{})
+	vsphere_secrets := []string{"vsphere-config-secret", "csi-vsphere-config"}
+	var secret *k8sv1.Secret
+	for _, vsphere_secret := range vsphere_secrets {
+		secret, err = secretApis.Get(vsphere_secret, metav1.GetOptions{})
+		if err == nil {
+			logger.Infof("Retrieved k8s secret, %s", vsphere_secret)
+			break
+		}
+		logger.WithError(err).Infof("Failed to get k8s secret, %s", vsphere_secret)
```

Comment:
Warnf

Reply:
We expect only one of the vsphere-secrets to be correct in any case. So should we treat the ones we have checked so far and rejected as warnings? It is expected behavior and nothing is really going wrong.